### ISSUE TITLE
Unexpected behavior when dealing with Protocol generics

### ISSUE THREAD

**Issue by BaconPancakes** [pylint-dev/pylint#9863]
### Bug description
Minimal Repro, see inline comments for false positive descriptions

```python

"""My module docstring."""
from collections.abc import Sized
from typing import Protocol, TypeVar

from typing_extensions import override, TypeAlias

# pylint: disable=too-few-public-methods

T = TypeVar("T")
V = TypeVar("V")


class MyClass(Sized, Protocol[T, V]):
    """My class docstring."""

    def my_method(self):
        """My docstring."""
        ...  # unnecessary-ellipse, but this is used as implementation "placeholder"


# unsubscriptable-object, but inherits from Generic Protocol
class MySubclass(MyClass[T, V]):
    """My class docstring."""

    def __len__(self) -> int:
        return 0

    @override
    def my_method(self):  # missing-function-docstring, but overrides MyClass
        return


# --------------------------------------------------------------------------- #
class AnotherClass(Protocol[T, V]):
    """My class docstring."""


MyAlias: TypeAlias = AnotherClass[T, int]

foobar = MyAlias[
    bool
]  # unsubscriptable-object, but just aliased the generic type variable above
```

### Command used

```shell
pylint
```

### Pylint output

```python
************* Module pylint_bugs
sandbox/pylint_bugs.py:18:8: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)
sandbox/pylint_bugs.py:22:17: E1136: Value 'MyClass' is unsubscriptable (unsubscriptable-object)
sandbox/pylint_bugs.py:29:4: C0116: Missing function or method docstring (missing-function-docstring)
sandbox/pylint_bugs.py:40:9: E1136: Value 'MyAlias' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

**Comment by nedbat** [comment]
This issue is labelled "Discussion", but if there's a discussion about it, I don't know where it is.

It would be great to add some subtlety to this check. Don't flag these ellipses if the class is a Protocol. pyright complains about a Protocol like this:
```python
class MyProtocol(Protocol):
    def how_many(self) -> int:
        """How many?"""
```
It complains that the return is None, but is declared as int.  pyright stops complaining if the body `...` is added, but then pylint is unhappy.

**Comment by Pierre-Sassoulas** [comment]
Sorry, implicitly I'm telling I'm not sure if we should do it by adding the discussion label. If it's incompatible with pyright we should fix it imo.
